tests/test_applications.py

```diff
@@ -486,3 +489,45 @@ async def startup():
 
         app.on_event("startup")(startup)
         assert len(record) == 1
+
+
+def test_middleware_stack_init(test_client_factory: Callable[[ASGIApp], httpx.Client]):
```

Comment:
This would be the first time using `httpx.Client` on this annotation on the test suite :eyes: 
```suggestion
def test_middleware_stack_init(test_client_factory: Callable[[ASGIApp], TestClient]):
```

Reply:
Hmm yeah I should have used `TestClient`. Luckily it's just a type annotation.